### ISSUE TITLE
Bug with large (Two decimals places) UID during a reverse mutation with Facets

### ISSUE THREAD

**Issue by MichelDiz** [dgraph-io/dgraph#1892]
Ref: https://discuss.dgraph.io/t/i-would-like-to-report-something-that-looks-like-a-bug-to-me/2008

- What version of Dgraph are you using?

Dgraph version   : v0.9.4
Commit SHA-1     : 779fc787
Commit timestamp : 2017-12-04 17:17:23 +1100
Branch           : HEAD

- Have you tried reproducing the issue with latest release?
Yes

I did the tests with JS Client and now (minutes ago) I did a test through HTTP Raw using Fetch()

  ```js
try {
    let response = await fetch(
      "http://localhost:8081/mutate",
      {
        method: "POST",
        headers: { "X-Dgraph-CommitNow": "true" },
        dataType: "json",
        body: data
      })
      .then((response) => response.json())
      .then((responseJson) => {
        console.log(`responseJson: \n${JSON.stringify(responseJson, null, 2)} <===========`);
      })
  } catch (errors) {
    console.log(`error: \n${JSON.stringify(errors, null, 2)}  <=========== <===========`);
  }
}
```

- What is the hardware spec (RAM, OS)?

Windows 10 Pro 1709
16GB RAM
FX8320E

Client:
 Version:      17.10.0-ce
 API version:  1.33
 Go version:   go1.8.3
 Git commit:   f4ffd25
 Built:        Tue Oct 17 19:00:02 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.11.0-ce
 API version:  1.34 (minimum version 1.12)
 Go version:   go1.8.5
 Git commit:   1caf76c
 Built:        Mon Nov 20 18:39:28 2017
 OS/Arch:      linux/amd64
 Experimental: false

- Steps to reproduce the issue (command/config used to run Dgraph).
https://youtu.be/LWVa5bxPUHM

- Expected behaviour and actual result.

Mutate and reverse edge with facets with UID greater than two decimals places

```json
responseJson:
{
  "errors": [
    {
      "code": "ErrorInvalidRequest",
      "message": "Uid: [10002] cannot be greater than lease: [10000]"
    }
  ],
  "data": null
}
```

**Comment by peterstace** [comment]
From looking at the video you liked, this looks like expected behaviour to me.

When using UIDs in mutations, you should only ever give UIDs that dgraph has given you first. UIDs should only be assigned by dgraph itself, not by the user. If you want to create new nodes in mutations, you should use [blank nodes](https://docs.dgraph.io/query-language/#blank-nodes-and-uid) instead.

**Comment by MichelDiz** [comment]
But I'm first using real UIDs generated by Dgraph. Only one or another part (to "prove fast") from the video I improvised.

In the beginning Dgraph genarete UIDs like 0x1~. All UIDs until 0x999 works fine with this. All genarated by Dgraph. After cleaning all (dropAll) the Dgraph does not return to generate small UIDs. It starts from where it left off with the larger UIDs. And this error happens again and again.

In this case I can not use blank nodes. For the simple reason I want to link two nodes that represent existing things that will be connected (in the future). This environment I created is to simulate something real and this little problem is tying it up.

I would not open this issue by silly - Manish R Jain recommended to open it and I reluctantly didn't so, without first testing more.

**Comment by peterstace** [comment]
> All genarated by Dgraph.

Ah, I didn't realize this. In that case, you are using dgraph correctly but there must be a bug in dgraph.

To fix this problem, we need to be able to reproduce it. Can you make the problem occur from scratch? If so, can you list out the steps? (starting from a completely blank state). That would be really useful :)

**Comment by MichelDiz** [comment]
Tomorrow I will have a MacOs ready to hand. I will setup a new environment and test on it. So when it's over, give it back. I'll log all the steps.

**Comment by MichelDiz** [comment]
Report: I did the tests on Debian. Without Docker.
Everything went fine, I made video from start to finish. But it's kind of big. If you need to send YT pro.

I do not understand where the problem might be. I forced Dgraph to create hundreds of new UIDs. Until it gets big enough. I did exactly the same thing as before. The code is exactly the same.

One problem that occurred a lot was when doing repeated registrations at the same time the Dgraph aborted the requests. I did think 20 registrations at once from users and it aborted something like 5 of them. Every 20 mutations.

**Comment by MichelDiz** [comment]
I'm closing this issue. Well I think it was a bug caused by some pollution made by me. I did a lot and I think all of these tests have resulted in this bug. But it is recorded here for later.

I deleted the volumes created by Dgraph for Docker and maybe this has solved. I just forgot to check. My Windows machine where the error occurred is now normal. So that was it. But it is strange to have the possibility of this bug caused by exhaustive use.

My Debian Linux test did not involve Docker.

My hypotheses
> Some cache may have bugged because of the countless experiments I've done.

> Some remnant of information may have been mismatched in folders on the Docker volume.

> Someone got some part of the code in the mean time and unwittingly solved it. It is not plausible.

> My Docker is crazy.
